### PR TITLE
frio: fbrowser styling

### PR DESCRIPTION
**Pull request by rabuzarus**
* new fancy styling for the filebrowser in frio
* fbrowser can now switch between image and file mode
* When an album is selected it uploads the image into that album folder

* extend `wall_upload.php` to submit the album name

### CODE REVIEW

**Comment by Hypolite** [comment]
Nice graphic overhail!

However, I have a few remarks. On my computer, the root Image page is too high, the folder column is oddly narrow and the individual folder buttons aren't visually distinct enough.

friendica-frio-fbrowser-too-high

On the Image subfolder page, the images themselves aren't sufficiently distinct from each other when their background color is similar, especially since their size vary by row.

friendica-frio-fbrowser-indistinct-images

This confusion between images was already there in the current file browser, but if you're taking a stab at redesigning, I thought I'd bring this up.

**Comment by rabuzarus** [comment]
I will take a look. Do you think a image border would be better for visual distinction.
I have a question about the first picture you have posted. The picture of the time does look very high. Actually the justified Gallery row height should be be 80px. Can you please check the results?

**Comment by Hypolite** [comment]
After clearing my JS cache, I get the same visual result, the first image stays 487px high.

**Comment by Hypolite** [comment]
I've seen the small grey border and the padding, however on smaller thumbnails the top border get eaten.

friendica-frio-fbrowser-missing-top-border

You can see the huge first image on this screen after clearing my cache as well.

**Comment by rabuzarus** [comment]
@Hypolite this is strange.
The first picture shouldn't be that big. There seems to be an issue with justifiedGallery. I haven't this issue :-/

I don't want to add to much padding, because we need to save space on small mobile phones

My fbrowser
fbrowser

**Comment by Hypolite** [comment]
I suppose it is because you don't have a portrait image in your gallery, but I wouldn't know much more about it.

**Comment by rabuzarus** [comment]
Can please send me the picture ?

**Comment by rabuzarus** [comment]
OK I downloaded it from your wall album. Lets see what happens with this pic in my browser

**Comment by Hypolite** [comment]
My wall album? How do I find it?

**Comment by rabuzarus** [comment]
Hmm I don't know what the source of this issue is. I uploaded your picture and it behaves like it should
fbrowser1

**Comment by Hypolite** [comment]
I really wasn't expecting you to have the same issue right out of box. I suggest you download all the pictures on my wall album and upload them in a separate folder to replicate the conditions more accurately.

**Comment by rabuzarus** [comment]
> My wall album? How do I find it?

Photos page. Select the album "wall photos"

**Comment by rabuzarus** [comment]
OK. Maybe later if I find some time

**Comment by Hypolite** [comment]
> Photos page. Select the album "wall photos"

I see that for me, but I don't see it for you, is it expected?

**Comment by rabuzarus** [comment]
Wall photos should have the same permissions like the post where you have inserted the picture. If you don't have set the permissions to exclude me I can see them

**Comment by rabuzarus** [comment]
e.g. I can't see the second (with test in item body) and the third picture (the one of the comment box)

**Comment by Hypolite** [comment]
Can I send them all to any email address you control?

File: view/theme/frio/templates/jot.tpl [hunk 1]
```diff
@@ -52,6 +52,9 @@
 						<a class="jot-preview-lnk" id="jot-preview-lnk-mobile" onclick="previewActive(); return false;">{{$preview}}</a>
 					</li>
 					{{/if}}
+					<li role="menuitem">
+						<a id="jot-browser-lnk-mobile" onclick="fbrowserActive(); return false;">{{$browser}}</a>
```
Comment (Hypolite) [inline view/theme/frio/templates/jot.tpl]: `<button type="button">` please if there's no `href` attribute!

**Comment by Hypolite** [comment]
The border doesn't quite solve the image distinction issue if the image background color is comparable to the border color. I'd prefer a solution based on `box-shadow`.

File: view/theme/frio/templates/jot.tpl [hunk 2]
```diff
@@ -40,20 +40,20 @@
 					{{* mark the first list entry as active because it is the first which is active after opening
 					the modal. Changing of the activity status is done by js in jot.tpl-header *}}
 					<li role="menuitem" style="display: none;">
-						<a class="jot-text-lnk" id="jot-text-lnk-mobile" onclick="jotActive(); return false;">{{$message}}</a>
+						<a href="#profile-jot-wrapper" class="jot-text-lnk" id="jot-text-lnk-mobile" onclick="jotActive(); return false;">{{$message}}</a>
```
Comment (annando) [inline view/theme/frio/templates/jot.tpl]: ```<button>``` would be recommend instead of these links, I think.

**Comment by rabuzarus** [comment]
Some notes regarding buttons:
Have a look at https://getbootstrap.com/components/#btn-dropdowns and https://getbootstrap.com/components/#nav

I want to use bootstrap not rewrite bootstrap

**Comment by rabuzarus** [comment]
> But I don't see how it is rewriting Bootstrap?

Try to replace link with buttons. They behave the same because they are intended to change stuff when they are clicked. Visually they are very different. A button is a box around text. A link is just clickable text. BS did write some css to easily create standard elements. If you you want it in a different way then you have to write your own css or if you want to another behavior/look than the bs style you have to overwrite it.
Try to replace the jot tab links to buttons. You can't do this easily. You have to change the standard button look to have the former look. Instead of just adding a bs class you need to do with buttons what bs did with links (+ you need to make sure that there are no regressions in other parts where buttons should look like real buttons) . This is what I mean by "rewriting bs".

**Comment by annando** [comment]
I just referred to this discussion: http://stackoverflow.com/questions/37981349/is-an-empty-element-such-as-a-href-valid-for-accessibility-purposes

If changing this wouldn't be a very big problem it would be great. Otherwise it would be good if these links could get roles for the accessibility.

**Comment by Hypolite** [comment]
> Try to replace the jot tab links to buttons. You can't do this easily. You have to change the standard button look to have the former look. Instead of just adding a bs class you need to do with buttons what bs did with links (+ you need to make sure that there are no regressions in other parts where buttons should look like real buttons) . This is what I mean by "rewriting bs".

Okay, that's exactly where I stopped because I would have had to adapt way more CSS from Bootstrap than I was ready too.

Otherwise, I'm following what @annando is referring about.

File: view/theme/frio/js/filebrowser.js [hunk 3]
```diff
@@ -140,61 +131,75 @@ var FileBrowser = {
...
+
+			FileBrowser.loadContent(url);
+		});
 	},
 
+	// Initialize the AjaxUpload for the upload buttons
 	uploadButtons: function() {
 		if ($("#upload-image").length) {
+			// To get the albumname we need to convert it from hex
+			var albumname = hex2bin(FileBrowser.folder);
```
Comment (Hypolite) [inline view/theme/frio/js/filebrowser.js]: Where is the `hex2bin` function defined? I'm getting the following error:
````
Uncaught ReferenceError: hex2bin is not defined
    at Object.uploadButtons (filebrowser.js:165)
    at Object.postLoad (filebrowser.js:233)
    at Object.init (filebrowser.js:92)
    at HTMLScriptElement.jsbrowser [as onreadystatechange] (modal.js:153)
````

**Comment by Hypolite** [comment]
Just tried the new changes, thanks for adding the shadow, however it's barely distinguishable because images are so close to each other. Here's the visual settings that worked the best for me:

`margin: 5` in filebrowser.js:256
and
`box-shadow: 2px 2px 5px 0px rgba(0, 0, 0, 0.2)` in style.css:1371

File: view/theme/frio/js/filebrowser.js (hunk 3 above)
Comment (rabuzarus) [inline view/theme/frio/js/filebrowser.js]: it is in theme.js

**Comment by rabuzarus** [comment]
Compromise: I set the margin to 4

**Comment by rabuzarus** [comment]
I uploaded your pictures and I could reproduce the bug that the first image is too big. I guess this is a bug in justified-Gallery :-/

File: view/theme/frio/js/filebrowser.js (hunk 3 above)
Comment (rabuzarus) [inline view/theme/frio/js/filebrowser.js]: maybe a browser cache refresh problem. I don't get this error

**Comment by Hypolite** [comment]
Well, your last margin change fixed my huge image issue!

**Pull request merged**